### DISCUSSION TITLE
Regex validation for squash merge commit message

### DISCUSSION THREAD

**Discussion by adamhamlin** [community/community#8688] [Pull Requests]
Hi,

It would be very nice to be able to prescribe a regex somewhere (branch protection rules?) that would be run against the content of a squash merge commit message. It would block the merge if the regex doesn't pass.

For example, this could be used to enforce [Conventional Commits](https://www.conventionalcommits.org) syntax on squash merges. There are lots of tools to enforce this syntax at various other steps in the process (e.g., PR title, PR body), but not ultimately for squash merge.

Thanks!

**Comment by mnojek** [comment]
It would help if there will be an action/workflow that can be triggered after clicking "Squash and merge" but before actually merging the PR. It would need to be an intermediate step which will prevent the PR from being merged when the checks do not pass, and it would need to revert the "Squash and merge" button to be enabled again. I think the [Merge Queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) feature may be a way to do something similar, but it's not yet in public beta.
